src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -629,191 +602,85 @@ public TypeSymbolWithAnnotations SetUnknownNullabilityForReferenceTypes()
-            {
-                return this;
-            }
-
-            public override TypeSymbolWithAnnotations AsNotNullableReferenceType()
-            {
-                return new WithoutCustomModifiers(_typeSymbol);
-            }
-
-            public override string ToDisplayString(SymbolDisplayFormat format)
-            {
-                return _typeSymbol.ToDisplayString(format) + "?";
-            }
-        }
-
-        // Rename since it's not necessarily a reference type.
-        private class ReferenceTypeUnknownNullabilityWithoutCustomModifiers : TypeSymbolWithAnnotations
-        {
-            protected readonly TypeSymbol _typeSymbol;
-
-            public ReferenceTypeUnknownNullabilityWithoutCustomModifiers(TypeSymbol typeSymbol)
-            {
-                Debug.Assert((object)typeSymbol != null);
-                Debug.Assert(!typeSymbol.IsNullableType());
-                _typeSymbol = typeSymbol;
-            }
-
-            public sealed override TypeSymbol TypeSymbol => _typeSymbol;
-            public sealed override bool? IsNullable => null;
-            public override ImmutableArray<CustomModifier> CustomModifiers => ImmutableArray<CustomModifier>.Empty;
-
-            public override TypeSymbol AsTypeSymbolOnly()
-            {
-                return _typeSymbol;
-            }
-
-            public sealed override bool Is(TypeSymbol other) => _typeSymbol.Equals(other, TypeCompareKind.CompareNullableModifiersForReferenceTypes);
-
-            public override TypeSymbolWithAnnotations WithModifiers(ImmutableArray<CustomModifier> customModifiers)
-            {
-                if (customModifiers.IsDefaultOrEmpty)
-                {
-                    return this;
-                }
-
-                return new ReferenceTypeUnknownNullabilityWithCustomModifiers(_typeSymbol, customModifiers);
-            }
-
-            protected sealed override TypeSymbolWithAnnotations DoUpdate(TypeSymbol typeSymbol, ImmutableArray<CustomModifier> customModifiers)
-            {
-                return CreateWithUnknownNullability(typeSymbol, customModifiers);
-            }
-
-            public override TypeSymbolWithAnnotations AsNullableReferenceType()
-            {
-                return new NullableReferenceTypeWithoutCustomModifiers(_typeSymbol);
-            }
-
-            public override TypeSymbolWithAnnotations AsNotNullableReferenceType()
-            {
-                return new WithoutCustomModifiers(_typeSymbol);
-            }
-
-            public override string ToDisplayString(SymbolDisplayFormat format)
-            {
-                return _typeSymbol.ToDisplayString(format);
-            }
-        }
-
+        /// <summary>
+        /// Nullable type parameter. The underlying TypeSymbol is resolved
+        /// lazily to avoid cycles when binding declarations.
+        /// </summary>
         private sealed class LazyNullableType : TypeSymbolWithAnnotations
```

Comment:
>LazyNullableType [](start = 29, length = 16)

It seems a bit odd that the non-lazy class is nested and the lazy one is not nested. I would expect them to be the same (nested seems better).

Reply:
Both lazy and non-lazy implementations are nested within `TypeSymbolWithAnnotations`.

---
In reply to: [185653319](https://github.com/dotnet/roslyn/pull/26528#discussion_r185653319) [](ancestors = 185653319)